src/utils.ts

```diff
@@ -1,32 +1,95 @@
+import multiformats, { CID } from "multiformats/basics.js";
+import dagcbor from "@ipld/dag-cbor";
+import dagjson from "@ipld/dag-json";
+import create from "@ipld/block";
+import base58 from "multiformats/bases/base58.js";
+
+multiformats.multibase.add(base58);
+multiformats.multicodec.add([dagcbor, dagjson]);
+const _Block = create(multiformats);
+
 // Basic constructor interface
-interface Constructore<T> {
-  new (parent: any): T
+interface Constructor<T> {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  new (parent: any): T;
 }
 
 /**
- * Magic function to lazily define child classes as namesapces on parent clases in a type-safe way.
+ * Magic function to lazily define child classes as namespaces on parent classes in a type-safe way.
  *
  * @param obj The object to which to add the subclass.
  * @param name The name of the property that subclass will be available under.
  * @param ctor The actual subclass to add.
  * @protected
  * @ignore
  */
-export function addSubclass<T>(obj: any, name: string, ctor: Constructore<T>) {
+export function addSubclass<T>(
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any,@typescript-eslint/explicit-module-boundary-types
+  obj: any,
+  name: string,
+  ctor: Constructor<T>
+): void {
   Object.defineProperty(obj.prototype, name, {
     // Properties should show up during enumeration of properties on Peer
     enumerable: true,
-    get: function() {
+    get: function () {
       // One-time lazily 'get' a new Bitswap and then cache it for subsequent calls
-      const value = new ctor(this)
+      const value = new ctor(this);
       // Now set the 'cached' value 'permanently'
       Object.defineProperty(this, name, {
         value: value,
         configurable: false,
         writable: false,
         enumerable: true,
-      })
-      return value
+      });
+      return value;
     },
-  })
+  });
+}
+
+export type Codec = never;
+
+export interface Reader<T> {
```

Comment:
We define the Block API types in here, and also wrap in some "custom" codec settings for ease of use elsewhere.